Source/JavaScriptCore/wasm/js/JSToWasm.cpp

```diff
@@ -214,6 +218,294 @@ void marshallJSResult(CCallHelpers& jit, const TypeDefinition& typeDefinition, c
+            auto gigacageConfig = bitwise_cast<uint8_t*>(&WebConfig::g_config) + static_cast<ptrdiff_t>(Gigacage::startOffsetOfGigacageConfig);
+            auto gigacageDisablingPrimitiveIsForbidden = gigacageConfig + OBJECT_OFFSETOF(Gigacage::Config, disablingPrimitiveGigacageIsForbidden);
+            jit.load8(gigacageDisablingPrimitiveIsForbidden, GPRInfo::regWS0);
+            auto doCaging = jit.branch32(CCallHelpers::NotEqual, GPRInfo::regWS0, MacroAssembler::TrustedImm32(0));
+            jit.load8(&Gigacage::disablePrimitiveGigacageRequested, GPRInfo::regWS0);
+            auto done = jit.branch32(CCallHelpers::NotEqual, GPRInfo::regWS0, MacroAssembler::TrustedImm32(0));
+            doCaging.link(&jit);
+            // cage()
+            auto basePtr = gigacageConfig + OBJECT_OFFSETOF(Gigacage::Config, basePtrs) + Gigacage::offsetOfPrimitiveGigacageBasePtr;
+            auto mask = Gigacage::primitiveGigacageMask;
+            jit.loadPtr(basePtr, GPRInfo::regWS0);
+            auto cageDone = jit.branch64(CCallHelpers::Equal, GPRInfo::regWS0, MacroAssembler::TrustedImm32(0));
+            jit.andPtr(MacroAssembler::TrustedImmPtr(mask), GPRInfo::wasmBaseMemoryPointer);
+            jit.addPtr(GPRInfo::regWS0, GPRInfo::wasmBaseMemoryPointer);
+            cageDone.link(&jit);
+            done.link(&jit);
+#endif
+#endif
+        }
+
+#if CPU(ARM64) || CPU(ARM64E)
+        jit.loadPair64(CCallHelpers::stackPointerRegister, MacroAssembler::TrustedImm32(0 * 8), GPRInfo::regWA0, GPRInfo::regWA1);
+        jit.loadPair64(CCallHelpers::stackPointerRegister, MacroAssembler::TrustedImm32(2 * 8), GPRInfo::regWA2, GPRInfo::regWA3);
+        jit.loadPair64(CCallHelpers::stackPointerRegister, MacroAssembler::TrustedImm32(4 * 8), GPRInfo::regWA4, GPRInfo::regWA5);
+        jit.loadPair64(CCallHelpers::stackPointerRegister, MacroAssembler::TrustedImm32(6 * 8), GPRInfo::regWA6, GPRInfo::regWA7);
+#elif USE(JSVALUE64)
+        jit.load64(CCallHelpers::Address { GPRInfo::regWA0, 0*8 }, CCallHelpers::stackPointerRegister);
+        jit.load64(CCallHelpers::Address { GPRInfo::regWA1, 1*8 }, CCallHelpers::stackPointerRegister);
+        jit.load64(CCallHelpers::Address { GPRInfo::regWA2, 2*8 }, CCallHelpers::stackPointerRegister);
+        jit.load64(CCallHelpers::Address { GPRInfo::regWA3, 3*8 }, CCallHelpers::stackPointerRegister);
+        jit.load64(CCallHelpers::Address { GPRInfo::regWA4, 4*8 }, CCallHelpers::stackPointerRegister);
+        jit.load64(CCallHelpers::Address { GPRInfo::regWA5, 5*8 }, CCallHelpers::stackPointerRegister);
+#else
+        jit.loadPair32(CCallHelpers::stackPointerRegister, MacroAssembler::TrustedImm32(0*8), GPRInfo::regWA0, GPRInfo::regWA1);
+        jit.loadPair32(CCallHelpers::stackPointerRegister, MacroAssembler::TrustedImm32(1*8), GPRInfo::regWA2, GPRInfo::regWA3);
+#endif
+
+#if CPU(ARM64) || CPU(ARM64E)
```

Comment:
You can also use this code for CPU(X86_64).

Reply:
macOS X86 build breaks when I use `loadPair64`; it's not defined for that architecture.